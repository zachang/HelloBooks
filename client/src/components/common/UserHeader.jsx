import React from 'react';
import {Link, browserHistory} from 'react-router';
import jwt_decode from 'jwt-decode';

export default class UserHeader extends React.Component {
  constructor() {
    super();
    this.logOut = this.logOut.bind(this);
  }

  logOut() {
    window.sessionStorage.removeItem('token');
    browserHistory.push('/');
  }

  componentDidMount() {
    $('.collapsible').collapsible();
    $('.dropdown-button').dropdown();
  }


  render() {
    const userDetails = jwt_decode(window.sessionStorage.token);
    return (
      <div className='col m10 l10 s12 right' style={{padding: '0px'}}>
        <div className='navbar-fixed'>
          <nav className='mainNav black col m10 l10' role='navigation'>
            <div className='nav-wrapper container'><Link id='logo-container' to='/user' className='brand-logo mainLogo'><img
              src='./imgs/hello.png' alt='hellobooks'/></Link>
              <ul className='user hide-on-small-and-down show-on-medium-and-up'>
                <Link className='dropdown-button btn black btn-prof' to='' data-activates='dropdown1'>
                  <i className='material-icons mat-icon right' style={{margin: '0% 0% 0% 1%'}}>arrow_drop_down</i>
                  {userDetails.username}
                </Link>
              </ul>

              <ul id='dropdown1' className='dropdown-content hide-on-small-and-down'>
                <li><Link to='/profile'>Profile</Link></li>
                <li className='divider'></li>
                <li><Link to='/returned'>Returned Books</Link></li>
                <li className='divider'></li>
                <li><Link to='/borrowed'>Borrowed Books</Link></li>
                <li className='divider'></li>
                <li><Link to=''>Change Password</Link></li>
                <li className='divider'></li>
                <li><Link to='#' onClick={this.logOut}>Logout</Link></li>
              </ul>

              <ul id='nav-mobile' className='side-nav grey darken-4 hide-on-med-and-up'>
                <li>
                  <div className='row walp'>
                  </div>
                </li>
                <li><Link to='/profile'>Profile</Link></li>
                <li className='no-padding'>
                  <ul className='collapsible collapsible-accordion'>
                    <li>
                      <Link className='collapsible-header'>Books Shelf<i className='material-icons mat-icon'>arrow_drop_down</i></Link>
                      <div className='collapsible-body'>
                        <ul>
                          <li><Link to='/user'>All Books</Link></li>
                          <li><Link to='/borrowed'>Borrowed Books</Link></li>
                          <li><Link to='/returned'>Returned Books</Link></li>
                        </ul>
                      </div>
                    </li>
                  </ul>
                </li>
                <li><Link to='#'>Change Password</Link></li>
                <li><Link to='#' onClick={this.logOut}>Logout</Link></li>
              </ul>
              <Link href='#' data-activates='nav-mobile' className='button-collapse menu-icon-link'><i
                className='material-icons hide-on-med-and-up'>menu</i></Link>
            </div>
          </nav>
        </div>
      </div>
    );
  }
}

