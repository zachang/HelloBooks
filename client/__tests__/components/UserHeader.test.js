import React from 'react';
import toJson from 'enzyme-to-json';
import { Router, browserHistory } from 'react-router';
import { shallow } from 'enzyme';

import UserHeader from '../../src/components/common/UserHeader.jsx';

const props = {
  profileImage: '../../src/build/imgs/eben2.jpg',
  profileUsername: 'Ebenezer'
};

const shallowWrapper = shallow(
  <Router history={browserHistory}>
    <UserHeader {...props}/>
  </Router>
);
const tree = toJson(shallowWrapper);
let wrapper;

describe('<UserHeader/>', () => {
  it('renders <UserHeader /> component', () => {
    expect(tree).toMatchSnapshot();
  });

  it('calls logOut', () => {
    wrapper = shallow(<UserHeader {...props} />);
    wrapper.find('.logOut').simulate('click');
  });
});

