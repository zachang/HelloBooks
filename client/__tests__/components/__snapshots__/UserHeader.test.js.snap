// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserHeader/> renders <UserHeader /> component 1`] = `""`;
